### PR TITLE
Drop SKU lib HTTP dependency, initialize billing navigation session

### PR DESCRIPTION
**Pull request by LukasPaczos**
Refs https://github.com/mapbox/mapbox-navigation-android/pull/2410. This PR removes the sku lib's dependency on `OkHttp` which allows us to include it in the `core` module and correctly start/stop billable navigation sessions.

### CODE REVIEW

File: libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ import com.mapbox.android.accounts.navigation.sku.v1.SkuGenerator
 import com.mapbox.android.accounts.navigation.sku.v1.TripsSku
 import com.mapbox.android.accounts.v1.AccountsConstants.MAPBOX_SHARED_PREFERENCES
 
-internal class MapboxNavigationAccounts private constructor() {
+class MapboxNavigationAccounts private constructor() {
```
Comment (Guardiola31337) [inline libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt]: This class should be under an `internal` package as it's not meant to be used by external developers. I'm going to cut a ticket to audit the code and do the same across all modules.

Comment (LukasPaczos) [reply libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt]: The only exposed methods are `startNavigation`, `stopNavigation` and `buildUrl`. They need to be public and we can't hide them if we want the SKU logic to live in a separate library.

Comment (Guardiola31337) [reply libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt]: Yeah, and that's 🆗 What I meant is that they should live under an `internal` package (and be `public` of course) so developers know that this class is meant to be internal and they shouldn't rely on it. That also give us the possibility to change APIs and break these internal APIs "without" breaking SemVer. It's a common practice in SDKs / libraries development 👀 https://github.com/square/okhttp/tree/master/okhttp/src/main/java/okhttp3/internal

Comment (Guardiola31337) [reply libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt]: Will cut a ticket describing ☝️ as I mentioned so we can follow up 🚀

Comment (LukasPaczos) [reply libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt]: I missed `package` word :upside_down_face:

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 2]
```diff
@@ -64,6 +62,9 @@ class MapboxNavigation(
                 ::paramsProvider
             )
         )
+        directionsSession.registerRouteObserver(createInternalRouteObserver())
+        directionsSession.registerRouteObserver(navigationSession)
```
Comment (abhishek1508) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: Where are we unregistering these observers registered inside `init`

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 3]
```diff
@@ -84,9 +84,8 @@ class MapboxNavigation(
             locationEngine,
             locationEngineRequest
         )
-
-        directionsSession.registerRouteObserver(createInternalRouteObserver())
         tripSession.registerOffRouteObserver(createInternalOffRouteObserver())
+        tripSession.registerStateObserver(navigationSession)
```
Comment (abhishek1508) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: Same as above

File: libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt [hunk 4]
```diff
@@ -48,4 +48,17 @@ internal class MapboxNavigationAccounts private constructor() {
...
+    fun buildResourceUrlWithSku(resourceUrl: String, querySize: Int): String {
+        val urlBuilder = StringBuilder(resourceUrl)
+        if (querySize == 0) {
+            urlBuilder.append("?")
+        } else {
+            urlBuilder.append("&")
+        }
+
+        val skuToken = obtainSkuToken()
+        urlBuilder.append("sku=$skuToken")
```
Comment (abhishek1508) [inline libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt]: Hardcoded `sku`

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 2 above)
Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: I'm going to cut a ticket to audit the code and check that all of the observers registered get unregistered. This is not only here 👍

File: libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt (hunk 4 above)
Comment (LukasPaczos) [reply libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt]: You have in mind replacing with a constant value?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 2 above)
Comment (LukasPaczos) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: We need to audit. This case should be safe though as the passed arguments and the sessions have the same lifecycle (all are created by `MapboxNavigation`).

Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: Follow up ticket https://github.com/mapbox/mapbox-navigation-android/issues/2418

File: libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt (hunk 1 above)
Comment (Guardiola31337) [reply libnavigation-sku/src/main/java/com/mapbox/navigation/sku/accounts/MapboxNavigationAccounts.kt]: Follow up ticket https://github.com/mapbox/mapbox-navigation-android/issues/2417

**Pull request merged**